src/ForwardedHeader.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace AspNetCore.ForwardedHttp
+{
+    //TODO: Need more efficient implementation
+    internal class ForwardedHeader
+    {
+        public ForwardedHeader(StringValues value)
+        {
+            for (var i = 0; i < value.Count; i++)
+            {
+                var header = value[i];
+                var sections = header.Split(";");
+
+                for (int z = 0; z < sections.Length; z++)
+                {
+                    var items = sections[z].Split(",");
+
+                    for (var j = 0; j < items.Length; j++)
+                    {
+                        var kv = items[j].Split("=");
+
+                        var key = kv[0].Trim();
+                        if (key.Equals("For"))
+                        {
+                            For.AddRange(kv[1].Split(","));
+                        }
+                        else if (key.Equals("By"))
+                        {
+                            By.AddRange(kv[1].Split(","));
+                        }
+                        else if (key.Equals("Host"))
+                        {
+                            Host.AddRange(kv[1].Split(","));
+                        }
+                        else if (key.Equals("Proto"))
+                        {
+                            Proto.AddRange(kv[1].Split(","));
+                        }
+                    }
+                }
+            }
+        }
+
+        public List<string> For { get; internal set; } = new List<string>();
```

Comment:
Splitting out the sections into flat lists like this breaks the spirit of the spec. The reason the segments are grouped like they are is so that you can tell which value was provided by which proxy. The values need to be kept together. E.g. 
```
class ForweardedValues
{
  string For { get; }
  string By { get; }
  string Host { get; }
  string Proto { get; }
}
IList<ForwardedValues> ParseProxyValues(StringValues header);
```

Reply:
How about this one? https://tools.ietf.org/html/rfc7239#section-7.5

> The HTTP request between the second proxy and the origin server
      has a "Forwarded: for=192.0.2.43,
      for=198.51.100.17;by=203.0.113.60;proto=http;host=example.com"
      header field.

how's the Forwarded header if it's passing from multiple proxies, and not every one apply all values?

Reply:
Right, that's why we need to know which ones provided which values. Some ForwardedValues properties may be null.

Reply:
For example we have three proxies and this is the header we receive. How can I know which proxy sent which values?
`for=10.10.10.1.0, for=20.20.20.20, for=30.30.30.30;by=8.8.8.8;proto=https;host=testhost, host=anothertesthost`

Reply:
https://tools.ietf.org/html/rfc7239#section-7.5
> The top-level list is represented as a list of HTTP header
   field-values as defined in Section 3.2 of [RFC7230].  The first
   element in this list holds information added by the first proxy that
   implements and uses this header field, and each subsequent element
   holds information added by each subsequent proxy.

The `,` divides the proxies. The `;` divides the proxy values.
`for=192.0.2.60;proto=http;by=203.0.113.43, for=192.0.2.43, for=198.51.100.17`
Proxy1: `for=192.0.2.60;proto=http;by=203.0.113.43`
Proxy2: `for=192.0.2.43`
Proxy3: `for=198.51.100.17`